decidim-core/lib/decidim/form_builder.rb

```diff
@@ -403,16 +403,17 @@ def datetime_field(attribute, options = {})
     def attachment(attribute, options = {})
       object_attachment = object.attachment.present?
       record = object_attachment ? object.attachment : object
-
       options = {
         titled: true,
         resource_class: "Decidim::Attachment",
         show_current: false,
         max_file_size: max_file_size(record, :file),
-        help: options[:help] || upload_help(record, attribute, options),
         label: I18n.t("decidim.forms.upload.labels.add_attachment"),
-        button_edit_label: I18n.t("decidim.forms.upload.labels.edit_image")
+        button_edit_label: I18n.t("decidim.forms.upload.labels.edit_image"),
+        extension_allowlist: Decidim.organization_settings(Decidim::Attachment).upload_allowed_file_extensions
       }.merge(options)
+      options[:help] = upload_help(record, attribute, options) if options[:help].blank?
```

Comment:
If the reason why you wanted to move this here is that the help text won't get an empty string, you could also use `.presence`:
https://www.rubydoc.info/docs/rails/Object:presence

So I believe you could achieve exactly the same by keeping the original but adding the `.presence` check to `options[:help]`:
```ruby
help: options[:help].presence || upload_help(record, attribute, options),
```

Reply:
I moved it because ```upload_help``` method uses ```extension_allowlist``` from the options. So we need to include ```extension_allowlist``` to the options before we can call ```upload_help```.

Reply:
Ah OK, thanks for the explanation!